src/DynamoCoreWpf/ViewModels/Core/WorkspaceViewModel.cs

```diff
@@ -363,57 +362,42 @@ public bool IsHomeSpace
 
+        [JsonIgnore]
+        internal string CurrentCheckSum { get; set; }
+
         /// <summary>
-        /// Returns the stringified representation of the connected nodes
+        /// Returns the stringified representation of the node connections in the workspace.
         /// </summary>
         [JsonIgnore]
         public string Checksum
         {
             get
             {
                 List<string> nodeInfoConnections = new List<string>();
-                JObject jsonWorkspace = JsonRepresentation;
-                var nodes = jsonWorkspace["Nodes"];                
+                var connectors = Connectors;
 
-                List<string> nodeIds = new List<string>();
-                foreach (JObject node in nodes)
+                foreach (var connector in Connectors)
                 {
-                    var nodeProperties = node.Children<JProperty>();
-                    JProperty id = nodeProperties.FirstOrDefault(x => x.Name == "Id");
-                    nodeIds.Add(id.Value.ToString());
-                }
-
-                nodeIds.Sort();
-
-                foreach (string nodeId in nodeIds)
-                {
-                    List<string> outputIds = new List<string>();
-                    var node = jsonWorkspace["Nodes"].Where(t => t.Value<string>("Id") == nodeId).Select(t => t).FirstOrDefault();
-                    var outputsProperty = node.Children<JProperty>().FirstOrDefault(x => x.Name == "Outputs");
-                    var outputs = (JArray)outputsProperty.Value;
-                    int outputIndex = 1;
-
-                    foreach (JObject output in outputs)
-                    {
-                        var outputProperties = output.Children<JProperty>();
-                        JProperty outputId = outputProperties.FirstOrDefault(x => x.Name == "Id");
-                        outputIds.Add(outputId.Value.ToString());
+                    var connectorModel = connector.ConnectorModel;
 
-                        var connectorsProperty = jsonWorkspace["Connectors"].Where(t => t.Value<string>("Start") == outputId.Value.ToString());
+                    var startingPort= connectorModel.Start;
+                    var endingPort = connectorModel.End;
 
-                        foreach (var connector in connectorsProperty)
-                        {
-                            var connectorProperties = connector.Children<JProperty>();
-                            JProperty endProperty = connectorProperties.FirstOrDefault(x => x.Name == "End");
-                            string inputId = (String)endProperty.Value;
+                    // node info connections has a unique id in the format: startnodeid[outputindex]endnodeid[outputindex].
+                    nodeInfoConnections.Add(startingPort.Owner.AstIdentifierGuid + "[" + startingPort.Index.ToString() + "]" + endingPort.Owner.AstIdentifierGuid + "[" + endingPort.Index.ToString() + "]");
```

Comment:
this is still bizarre to me - I guess you want to base your hash data only on the connections of the graph and no other properties, and in this case sha256 is a good has to choose as any small change in the input will create a big change in the output.

What I was imagining originally was that you could just use a hash that generates collisions - like SHA128 or MD5 - then just use the current JSON of the entire graph, but that is much less controllable than the solution you have here - depends on what you need exactly.